docs/resources/apigw_api_publishment.md

```diff
@@ -0,0 +1,94 @@
+---
+subcategory: "APIG"
+---
+
+# opentelekomcloud_apigw_api_publishment_v2
+
+Using this resource to publish an API to the environment or manage a historical publishing version within OpenTelekomCloud.
+
+~> If you republish on the same environment or switch versions through other ways (such as console) after the API is
+published through terraform, the current resource attributes will be affected, resulting in data inconsistency.
+
+## Example Usage
+
+### Publish a new version of the API
+
+```hcl
+variable "gateway_id" {}
+variable "environment_id" {}
+variable "api_id" {}
+
+resource "opentelekomcloud_apigw_api_publishment_v2" "default" {
+  gateway_id     = var.gateway_id
+  environment_id = var.environment_id
+  api_id         = var.api_id
+}
+```
+
+### Switch to a specified version of the API which is published
+
+```hcl
+variable "gateway_id" {}
+variable "environment_id" {}
+variable "api_id" {}
+variable "version_id" {}
+
+resource "opentelekomcloud_apigw_api_publishment_v2" "default" {
+  gateway_id     = var.gateway_id
+  environment_id = var.environment_id
+  api_id         = var.api_id
+  version_id     = var.version_id
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to publish APIs.
+  If omitted, the provider-level region will be used.
+  Changing this will create a new resource.
+
+* `instance_id` - (Required, String, ForceNew) Specifies an ID of the APIGW dedicated instance to which the API belongs
+  to. Changing this will create a new resource.
+
+* `environment_id` - (Required, String, ForceNew) Specifies the ID of the environment to which the current version of the API
+  will be published or has been published.
+  Changing this will create a new resource.
+
+* `api_id` - (Required, String, ForceNew) Specifies the ID of the API to be published or already published.
+  Changing this will create a new resource.
+
+* `description` - (Optional, String) Specifies the description of the current publish.
+
+* `version_id` - (Optional, String) Specifies the version ID of the current publish.
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID, which is constructed from the instance ID, environment ID, and API ID, separated by slashes.
+
+* `environment_name` - The name of the environment to which the current version of the API is published.
+
+* `published_at` - Time when the current version was published.
+
+* `publish_id` - The publishing ID of the API in current environment.
+
+* `history` - All publish history of the API.
+  The [object](#publishment_history) structure is documented below.
```

Comment:
I think we should either document structure without links to objects (like we did before) or refactor everything. Might be additional task for backlog

Reply:
Yes, another refactoring